extension-style/src/main/java/com/mapbox/maps/extension/style/layers/Layer.kt

```diff
@@ -182,14 +183,15 @@ abstract class Layer : StyleContract.StyleLayerExtension {
  * @return StyleLayerPlugin
  */
 fun StyleManagerInterface.getLayer(layerId: String): Layer? {
-  val expected = this.getStyleLayerProperties(layerId)
-  expected.value?.let { value ->
-    @Suppress("UNCHECKED_CAST")
-    val map = value.contents as HashMap<String, Value>
-    val source = map["source"]?.contents?.let { it as String }
-    val type = map["type"]?.contents?.let { it as String }
-    return when (type) {
+  val source =
+    if (layerId != "background" && layerId != "location-indicator" && layerId != "sky") {
```

Comment:
I would prefer to put all these `layerId` into a list and use `contains` function to do the check.